File: hasher.py
#! /usr/bin/env python
from hashlib import sha1

SIZE_128_MB = 131072
BIG_FILE_PATH = 'big_file'

def file_chunker(file_path, chunk_size=SIZE_128_MB):
    with open(file_path, 'rb') as big_file:
        for chunk in iter(lambda: big_file.read(chunk_size), ''):
            yield chunk

def get_sha1_hash(file_path):
    hash_obj = sha1()
    for chunk in file_chunker(file_path):
        hash_obj.update(chunk)
    return hash_obj.hexdigest()

if __name__ == '__main__':
    file_hash = get_sha1_hash(BIG_FILE_PATH)
    print file_hash

File: hasher.go
package main

import (
    "os"
    "io"
    "fmt"
    "log"
    "hash"
    "crypto/sha1"
)

const SIZE_128_MB = 128<<10
const BIG_FILE_PATH = "big_file"

var c chan string
var flag chan bool
var h hash.Hash
var wf *os.File

func file_chunker(file *os.File, chunk_size int){
    chunk := make([]byte, chunk_size)
    for {
        chunk = chunk[:cap(chunk)]
        n, err := file.Read(chunk)
        chunk = chunk[:n]
        if err == io.EOF {
            break
        }
        if err != nil {
            panic(err)   
        }
        c <- string(chunk)
    }
    close(c)
    return
}

func hasher() {
    h = sha1.New()
    for str := range c {
        h.Write([]byte(str))
    }
    flag <- true
    return
}

func main() {
    big_f, err := os.Open(BIG_FILE_PATH)
    if err != nil {
        log.Fatal(err)
    }
    defer big_f.Close()

    c = make(chan string, 20)
    flag = make(chan bool)

    go file_chunker(big_f, SIZE_128_MB)
    go hasher()
    <-flag
    fmt.Printf("%x\n", h.Sum(nil))
}